modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -984,6 +986,88 @@ impl LightningClientModule {
 
         Ok(state)
     }
+
+    /// Generate an lnurl for the client.
+    pub async fn generate_lnurl(
+        &self,
+        recurringd: Option<SafeUrl>,
+        gateway: Option<SafeUrl>,
+    ) -> Result<String, RegisterLnurlError> {
+        let recurringd = match recurringd {
+            Some(recurringd) => recurringd,
+            None => SafeUrl::from_str("https://recurringdv2.fedimint.org").unwrap(),
+        };
+
+        let gateways = if let Some(gateway) = gateway {
+            vec![gateway]
+        } else {
+            let gateways = self
+                .module_api
+                .gateways()
+                .await
+                .map_err(|e| RegisterLnurlError::FederationError(e.to_string()))?;
+
+            if gateways.is_empty() {
+                return Err(RegisterLnurlError::NoVettedGateways);
+            }
+
+            gateways
+        };
```

Comment:
What's the use case for only using one gateway? I think the main advantage of keeping `recurringd` vs putting it in the gateway is that multiple gateways can be used.